### PR TITLE
Update the 2FA page

### PR DESCRIPTION
**Pull request by tommytran732**

### CODE REVIEW

**Comment by opheron** [comment]
@tommytran732 Hey, just a heads up I recently submitted PR #861 which also makes significant changes the 2FA article.

File: docs/android.md [hunk 1]
```diff
@@ -354,8 +354,7 @@ We have these general tips:
...
  - Check if there are any differences between the F-Droid version and the Google Play Store version. Some applications like [IVPN](https://www.ivpn.net/) do not include certain features (eg [AntiTracker](https://www.ivpn.net/knowledgebase/general/antitracker-faq/)) in their Google Play Store build out of fear of censorship by Google.
-
-Evaluate whether the additional features in the F-Droid build are worth the slower updates. Also think about whether faster updates from the Google Play Store are worth the potential privacy isues in your [threat model](/threat-modeling/).
+ - Evaluate whether the additional features in the F-Droid build are worth the slower updates. Also think about whether faster updates from the Google Play Store are worth the potential privacy isues in your [threat model](/threat-modeling/).
```
Comment (dngray) [inline docs/android.md:357]: > isues

File: docs/multi-factor-authentication.md [hunk 2]
```diff
@@ -6,19 +6,48 @@ icon: 'material/two-factor-authentication'
...
 
-The idea behind 2FA is that even if a hacker is able to figure out your password (something you *know*), they will still need a device you own like your phone (something you *have*) in order to generate the code needed to log in to your account. 2FA methods vary in security based on this premise: The more difficult it is for an attacker to gain access to your 2FA method, the better. Examples of 2FA methods from strongest to weakest are Email or SMS codes, Push Notifications, Software (TOTP) Code-Generating Apps, and Hardware Keys.
+The idea behind 2FA is that even if a hacker is able to figure out your password (something you *know*), they will still need a device you own like your phone (something you *have*) in order to generate the code needed to log in to your account. 2FA methods vary in security based on this premise: The more difficult it is for an attacker to gain access to your 2FA method, the better. Examples of 2FA methods from strongest to weakest are Email codes, SMS codes, Push Notifications, Time-based One-time Password (TOTP), Yubico OTP, and FIDO2/U2F.
```
Comment (dngray) [inline docs/multi-factor-authentication.md:9]: I was pretty sure I edited this "strongest to weakest" thing, cos it sounds like email codes are the strongest and SMS codes are the second strongest.

File: docs/multi-factor-authentication.md [hunk 3]
```diff
@@ -6,19 +6,48 @@ icon: 'material/two-factor-authentication'
...
 
 ## MFA Method Comparison
 
-==**SMS Codes** or Emailed Codes are better than nothing at all, but only marginally.== Getting a code over SMS or Email takes away from the "something you *have*" idea, because there are a variety of ways a hacker could take over your phone number or gain access to your emails without having physical access to any of your devices at all!
+### Email or SMS 2FA
+==**Email** or **SMS 2FA** are better than nothing at all, but only marginally.== Getting a code over SMS or Email takes away from the "something you *have*" idea, because there are a variety of ways a hacker could take over your phone number or gain access to your emails without having physical access to any of your devices at all!
```
Comment (dngray) [inline docs/multi-factor-authentication.md:14]: Maybe we could use a link to https://en.wikipedia.org/wiki/SIM_swap_scam ?

File: docs/multi-factor-authentication.md [hunk 4]
```diff
@@ -6,19 +6,48 @@ icon: 'material/two-factor-authentication'
...
+To make matters worse, if a hacker gains access to your email, they typically would be able to use it to both reset your password and receive the second factor authentication code, giving them full access to your account. ==If you have to choose between SMS 2FA and Email 2FA, and if you do not mind giving out your phone number, SMS is preferable.==
```
Comment (dngray) [inline docs/multi-factor-authentication.md:16]: Generally this is why we would recommend 2FA on an email too.

**Comment by dngray** [comment]
Feedback on the MFA article, would be don't make it too verbose or complex, as it is an entry article.

File: docs/multi-factor-authentication.md (hunk 4 above)
Comment (tommytran732) [reply docs/multi-factor-authentication.md:16]: I will mention it briefly in the general recommendations section. I don't think it matters much though, the email is not completely in the user's control like say, a TOTP app or a Yubikey is. What if the email server gets hacked?

File: docs/multi-factor-authentication.md (hunk 3 above)
Comment (tommytran732) [reply docs/multi-factor-authentication.md:14]: How would you phrase it?

File: docs/multi-factor-authentication.md (hunk 2 above)
Comment (tommytran732) [reply docs/multi-factor-authentication.md:9]: Fixed.

File: docs/multi-factor-authentication.md [hunk 5]
```diff
@@ -68,6 +68,8 @@ If you have to use Email for 2FA, make sure that the email account itself is sec
 
     The **Yuibkey** was one is one of the most popular security keys. Some Yubikey models have a wide range of features such as for [Universal 2nd Factor (U2F)](https://en.wikipedia.org/wiki/Universal_2nd_Factor), [FIDO2 WebAuthn](https://en.wikipedia.org/wiki/WebAuthn), [Yubico OTP](https://developers.yubico.com/OTP/), [PIV](https://en.wikipedia.org/wiki/FIPS_201), [OpenPGP](https://developers.yubico.com/PGP/), [TOTP and HOTP](https://developers.yubico.com/OATH/) authentication. One of the benefits of these models is that one key can do most things.
 
+    The [comparison table](https://www.yubico.com/store/compare/) shows the features and how the Yubikeykeys compare. We highly recommend that you select keys from the Yubikey 5 Series.
```
Comment (elitejake) [inline docs/multi-factor-authentication.md:71]: ~~Yubikeykeys~~ to Yubikeys

**Review by elitejake — CHANGES_REQUESTED** [review]

**Comment by tommytran732** [comment]
> Feedback on the MFA article, would be don't make it too verbose or complex, as it is an entry article.

Do you wanna split it into 2 pages then?

**Comment by tommytran732** [comment]
IMO just recommending the Nitrokey and Yubikey and pretending that they are equal is not enough. Likewise, recommending hardware security keys over authenticator app is not great either. It depends heavily on which exact protocol is being used and how the secrets are stored.

**Comment by realguyman** [comment]
It should be noted that TOTP-based authentication is _still_ vulnerable to phishing attacks.

And it should be noted that hardware security keys and services which adopt FIDO2/U2F are not vulnerable to phishing attacks.

**Comment by realguyman** [comment]
If someone is going to purchase a hardware security key, it should be for FIDO2/U2F support.

Aegis Authenticator and other similar applications are secure enough for TOTP-based multi-factor authentication.

**Comment by tommytran732** [comment]
> If someone is going to purchase a hardware security key, it should be for FIDO2/U2F support.
> 
> Aegis Authenticator and other similar applications are secure enough for TOTP-based multi-factor authentication.

Depends. Yes, that is the most common use case, but it is not always applicable.
Take the Nitrokey storage 2 for example. It is an okay PGP smartcard, it has system integrity verification integration with heads (this is unique to Nitrokeys), and encrypted password storage. It stores HOTP and TOTP secrets as well, tho they are not encrypted.

For certain use cases, it is okay as a hardware security key despite of the fact that it does not support FIDO2/U2F.

File: docs/tools.md [hunk 6]
```diff
@@ -314,6 +314,19 @@ We [recommend](dns.md#why-should-i-use-encrypted-dns) a number of encrypted DNS
 
 [Learn more...](metadata-removal-tools.md)
 
+### Multi-Factor Authentication Tools
+
+<div class="grid cards" markdown>
+
+- ![YubiKeys](/assets/img/multi-factor-authentication/yubikey.png){ .twemoji } [YubiKey](https://www.yubico.com/)
+- ![Nitrokey](/assets/img/multi-factor-authentication/nitrokey.jpg){ .twemoji } [Nitrokey](https://www.nitrokey.com/)
```
Comment (jonaharagon) [inline docs/tools.md:322]: These images could be improved.

**Pull request merged**